### DISCUSSION TITLE
use_cuda Multi GPU train

### DISCUSSION THREAD

**Discussion by whzikaros** [coqui-ai/TTS#798] [General]
Why all train on one GPU when distributed training?

**Comment by erogol** [comment]
this explains how you should use multi-gpu training https://tts.readthedocs.io/en/latest/training_a_model.html

**Reply by whzikaros** [reply]
Thank you！But when I train on two GPUS, there is a bug: UnboundLocalError: local variable 'loader' referenced before assignment.
There is no this bug on single GPU. Why?

**Reply by whzikaros** [reply]
I found the reason，the code need alter.
In train.py function--_get_loader()
torch.nn.parallel.distributed.DistributedDataParallel hasn't attribute which is "get_data_loader",
So add code:
if isinstance(model, torch.nn.parallel.distributed.DistributedDataParallel):
    model = model.module

**Reply by erogol** [reply]
You better open an issue. I don't understand where the error is from the msg above.
